Ast/src/Lexer.cpp

```diff
@@ -660,6 +742,34 @@ Lexeme Lexer::readNext()
         }
     }
 
+    case '{':
+    {
+        consume();
+
+        braceStack.push_back(BraceType::Normal);
```

Comment:
There's an opportunity to optimize this a tiny bit, so that all scripts that don't have string interpolation syntax don't unnecessarily need this stack.

When the stack is empty or if the top of the stack is `BraceType::Normal`, `}` is a valid production. So you can do this:

```suggestion
        if (braceStack.empty())
            braceStack.push_back(BraceType::Normal);
```